platforms/jvm/code-quality/src/main/groovy/org/gradle/api/plugins/quality/Pmd.java

```diff
@@ -401,9 +328,9 @@ public Property<Boolean> getIncrementalAnalysis() {
      * @since 5.6
      */
     @LocalState
-    @ToBeReplacedByLazyProperty
-    public File getIncrementalCacheFile() {
-        return new File(getTemporaryDir(), "incremental.cache");
+    @ReplacesEagerProperty
+    public Provider<RegularFile> getIncrementalCacheFile() {
+        return incrementalCacheFile;
     }
```

Comment:
This potentially should just be a `RegularFileProperty` and either protected or package-private.

I don't remember if we tried it when this was added, but does `@LocalState` work on private methods?  This really has no reason to be exposed other than to add the annotation on this file path. If we could put `@LocalState` on a private method/field, this would just go away as a public API.

Reply:
It doesn't work with private method currently, you get:
```
In plugin 'org.gradle.api.plugins.quality.PmdPlugin_Decorated' type 'org.gradle.api.plugins.quality.Pmd' property 'incrementalCacheFile' is private and annotated with @LocalState.

Reason: Annotations on private getters are ignored.
```